organizations/data.py

```diff
@@ -327,39 +353,64 @@ def bulk_create_organization_courses(organization_course_pairs):
         in organization_course_pairs
     }
 
-    # Grab all existing (lowercased org short name, course key string) pairs from db,
-    # filtering for the ones requested for creation in `organization_course_pairs`, and
-    # indexing by db `id` of org_course object (we need this later for the bulk update).
-    orgslug_courseid_pairs_to_activate_by_id = {
-        org_course_id: (short_name.lower(), course_id)
-        for org_course_id, short_name, course_id
+    # Grab all existing (lowercased org short name, course key string, is_active) triples
+    # from db, filtering for the ones requested for creation in `organization_course_pairs`,
+    # and indexing by db `id` of org_course object (we need this later for the bulk update).
+    existing_orgslug_courseid_triples_by_id = {
+        org_course_id: (short_name.lower(), course_id, is_active)
```

Comment:
Nit: Once you start getting into 3+ sized tuples, having objects or `namedtuples` can help a lot with the readability.

Reply:
Agreed. Reading through this my code week later, the anonymous triples make this pretty hairy.